doc/developers/contributing.rst

```diff
@@ -455,9 +455,12 @@ complies with the following rules before marking a PR as ``[MRG]``. The
    functionality is useful in practice and, if possible, compare it to other
    methods available in scikit-learn.
 
-10. New features often need to be illustrated with narrative documentation in
-    the user guide, with small code snippets. If relevant, please also add
-    references in the literature, with PDF links when possible.
+10. New features have some maintenance overhead. It is expected that the person
+    who submits new code does part of the maintenance for that code at least
+    initially. New features need to be illustrated with narrative
```

Comment:
> at least initially

What would be the time frame?

Reply:
I don't think it is realistic to ask for a specific lapse of time, I'm assuming the good faith of the contributor here and that the review process will be able to highlight their good willing.